### PR TITLE
Toughness

### PR DESCRIPTION
**Pull request by wershlak**
I based this branch off the advanced offense branch cause there'd be a lot of conflicts otherwise. It should clean up once https://github.com/sneezymud/sneezymud/pull/274 is merged.

### CODE REVIEW

File: code/code/misc/combat.cc [hunk 1]
```diff
@@ -3982,6 +3989,9 @@ int TBeing::oneHit(TBeing *vict, primaryTypeT isprimary, TThing *weapon, int mod
 	}
       }
 
+  // Add toughness 
+  doToughness(vict);
```
Comment (cizra) [inline code/code/misc/combat.cc:3993]: indent

File: code/code/misc/combat.cc [hunk 2]
```diff
@@ -5772,3 +5782,55 @@ bool restrict_xp(const TBeing *caster, TBeing *victim, int duration)
   return true;
 }
 
+// called when the ch is hit and it should proc a "stack" of toughness
+void doToughness(TBeing *ch)
+{
+  int MAX_TOUGHNESS = 10;
+  affectedData *hjp;
```
Comment (cizra) [inline code/code/misc/combat.cc:5789]: Could we have a more descriptive name here? I have no idea what hjp means.

File: code/code/misc/info.cc [hunk 3]
```diff
@@ -1893,6 +1894,14 @@ sstring TBeing::describeAffects(TBeing *ch, showMeT showme) const
             describeDuration(this, aff->duration);
         }
         break;
+      case SKILL_TOUGHNESS:
+        if(show){
+          sstring noun = "stack";
+          if (aff->modifier2 > 1)
```
Comment (cizra) [inline code/code/misc/info.cc:1900]: Or you could do a ternary ?: and a single assignment.

File: code/code/misc/combat.cc (hunk 1 above)
Comment (wershlak) [reply code/code/misc/combat.cc]: Fixed it. The inconsistency of tab vs space in the code is annoying

**Comment by wershlak** [comment]
Rebased and fixed up those issues.

**Pull request merged**